### PR TITLE
Support Credit Card Closing Info

### PR DESCRIPTION
**Pull request by xiuzhong**

### CODE REVIEW

File: lib/ofx/parser/ofx220.rb [hunk 1]
```diff
@@ -7,6 +7,10 @@ def accounts_info
         @accounts_info ||= html.search('acctinfo').collect { |node| build_account_info(node) }
       end
 
+      def credit_card_closing_info
```
Comment (ouranos) [inline lib/ofx/parser/ofx220.rb]: This is present in the OFX 1.0.2 specs so you should put this method in the 1.0.2 parser

**Review by ouranos — CHANGES_REQUESTED** [review]
Code looks good to me (a part from the version) but `staging` is an integration branch so we shouldn't branch off of it.

Look at how the other branches are structured:
- branched off of `master`
- PR open against the upstream repo
- Pulled into `staging` while waiting for the PR to be merged

So you should branch off of `master` and open a PR on the upstream repo  (annacruz), then ping me and I'll pull it in our `staging` branch

File: lib/ofx/parser/ofx102.rb [hunk 2]
```diff
@@ -187,9 +191,33 @@ def build_available_balance(node)
...
+          last_payment_amount: to_decimal_or_nil(nested_closing.search('lastpmtinfo > lastpmtamt').inner_text),
+        })
+      end
+
       def to_decimal(amount)
         BigDecimal.new(amount.to_s.gsub(',', '.'))
       end
+
+      def to_decimal_or_nil(amount)
+        amount.nil? || amount.empty? ? nil : to_decimal(amount)
```
Comment (iseessel) [inline lib/ofx/parser/ofx102.rb:215]: can you use `amount.present?`:

`amount.present? ? to_decimal(amount) : nil`

Comment (iseessel) [reply lib/ofx/parser/ofx102.rb:215]:
```
2.3.4 :003 > ''.present?
 => false
2.3.4 :004 > [].present?
 => false
2.3.4 :005 > nil.present?
 => false
2.3.4 :006 > false.present?
 => false
```

File: lib/ofx/parser/ofx102.rb [hunk 3]
```diff
@@ -187,9 +191,33 @@ def build_available_balance(node)
...
+      end
+
+      def build_date_or_nil(date)
+        date.nil? || date.empty? ? nil : build_date(date)
```
Comment (iseessel) [inline lib/ofx/parser/ofx102.rb:219]: same as above.

**Review by iseessel — APPROVED** [review]
LGTM -- just one suggestion.

File: spec/ofx/credit_card_closing_info_spec.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      before do
+        @credit_card_closing_info = @parser.credit_card_closing_info[0]
+      end
+
+      it "should contain the correct values" do
+        @credit_card_closing_info.account_id.should == '4400720'
+        @credit_card_closing_info.currency_default.should == 'USD'
+        @credit_card_closing_info.fit_id.should == 'FITIDCCSTMT1399918849'
+        @credit_card_closing_info.date_close.should == Time.gm(2018, 6, 4)
+        @credit_card_closing_info.opening_balance.should == nil
```
Comment (iseessel) [inline spec/ofx/credit_card_closing_info_spec.rb:35]: Can we add a spec with a non `nil` opening balance?

**Comment by ouranos** [comment]
Included in #5

**Pull request closed without merging**